pkg/proxy/mark.go

```diff
@@ -14,32 +14,40 @@
 
 package proxy
 
-// Magic markers are attached to each packet. The lower 16 bits are used to
-// identify packets which have gone through the proxy and to determine whether
-// the packet is coming from a proxy at ingress or egress, or from the host.
-// The marking is compatible with Kubernetes's use of the packet mark.  The
-// upper 16 bits can be used to carry the security identity.
+// Magic markers are attached to each packet.
+//
+//  1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2
+// +-----------------------------+-----------+-------+-------------+
+// | identity-A                  | k8s       | mark  | identity-B  |
+// +-----------------------------+-----------+-------+-------------+
+//
+// identity-A: [ identity & 0xFFFF ]
+// identity-B: [ (identity >> 16) & 0xFF ]
```

Comment:
Hmm. I found these comments very confusing compared to the diagram above and the mask definitions below. Maybe I don't understand what `identity` is supposed to be here.

Why isn't it the below?

```
identity-A: [ identity >> 16 ]
identity-B: [ identity & 0xFF ] 
```

Reply:
Will improve

Reply:
Fixed

Reply:
Much clearer, thanks!